deploy/deployUtils.ts

```diff
@@ -245,12 +245,18 @@ export async function deployMetaswapPools(
       basePoolName!,
       metaPoolName,
     )
-    // verify contract
+    // verify contracts
     await verifyContract(hre, metaPoolName)
-  }
-  // register new pools
-  if (newDeploypools.length > 0) {
-    await registerPools(hre, newDeploypools)
+    await verifyContract(hre, lpTokenName)
+
+    // register new pools
+    if (newDeploypools.length > 0) {
+      await registerPools(hre, newDeploypools)
+      // If new pools require a gauge deploy
+      if (pool.deployGauge) {
+        deployLiquidityGauge(hre, lpTokenName, lpTokenAddress)
+      }
+    }
```

Comment:
Shouldnt `registerPools()` call be outside the for loop at L158?
Or else this gets called `newDeploypools.length` times